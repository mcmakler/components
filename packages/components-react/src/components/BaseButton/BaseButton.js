import React from 'react';
import PropTypes from 'prop-types';

/** BaseButton component description imported from comments inside the component file */
const BaseButton = ({ disabled, label, onClick }) => (
  <button className="button" disabled={disabled} onClick={onClick}>
    {label}
  </button>
);

BaseButton.defaultProps = {
  disabled: false,
  onClick: () => {},
};

BaseButton.propTypes = {
  /** Boolean indicating whether the button should render as disabled */
  disabled: PropTypes.bool,
  /** button label. */
  label: PropTypes.string.isRequired,
  /** onClick handler */
  onClick: PropTypes.func,
};

export default BaseButton;
