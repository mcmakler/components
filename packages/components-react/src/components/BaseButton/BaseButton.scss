@import "~mcmakler-components-style/lib/buttons";
