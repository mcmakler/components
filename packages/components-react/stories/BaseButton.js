import React from 'react';
import { action, storiesOf } from '@storybook/react';
import { BaseButton } from '../src';
import '../src/components/BaseButton/BaseButton.scss';

storiesOf('BaseButton', module)
  .addWithInfo(
    'enabled',
    `
      This is Base button
    `,
    () => (
      <BaseButton
        onClick={action('clicked')}
        label="Standard Button"
      />
    ),
  );
