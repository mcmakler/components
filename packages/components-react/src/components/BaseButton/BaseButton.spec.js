import React from 'react';
import renderer from 'react-test-renderer';
import BaseButton from './BaseButton';

test('Correctly renders.', () => {
  const label = 'Render Test';
  const tree = renderer.create(<BaseButton label={label} />).toJSON();

  expect(tree).toMatchSnapshot();
});
