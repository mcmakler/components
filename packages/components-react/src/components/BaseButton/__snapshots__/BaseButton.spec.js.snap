// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders. 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
>
  Render Test
</button>
`;
